flang/include/flang/Parser/parsing.h

```diff
@@ -59,6 +62,7 @@ class Parsing {
   const SourceFile *Prescan(const std::string &path, Options);
   void EmitPreprocessedSource(
       llvm::raw_ostream &, bool lineDirectives = true) const;
+  void EmitPreprocessorMacros(llvm::raw_ostream &) const;
```

Comment:
Why are you dumping the macros from the parser?  Their definitions all come from the driver via Fortran::parser::Options::predefinitions -- except `_CUDA` which really should come from the drivers anyway.  Print them from the drivers instead.

Reply:
Ah, I see; you also want to dump macros that are defined in headers; never mind.

Reply:
We need to print macros that are defined in the source code.